assets/js/components/UserMenu.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * UserMenu component.
+ *
+ * Site Kit by Google, Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { Fragment, useState, useRef, useEffect, useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { getSiteKitAdminURL, clearWebStorage } from '../util';
+import Dialog from './dialog';
+import Button from './button';
+import Menu from './menu';
+import Modal from './modal';
+import { STORE_NAME as CORE_SITE } from '../googlesitekit/datastore/site/constants';
+import { STORE_NAME as CORE_USER } from '../googlesitekit/datastore/user/constants';
+
+const { useSelect } = Data;
+
+function UserMenu() {
+	const proxyPermissionsURL = useSelect( ( select ) => select( CORE_SITE ).getProxyPermissionsURL() );
+	const userEmail = useSelect( ( select ) => select( CORE_USER ).getEmail() );
+	const userPicture = useSelect( ( select ) => select( CORE_USER ).getPicture() );
+	const [ dialogActive, toggleDialog ] = useState( false );
+	const [ menuOpen, toggleMenu ] = useState( false );
+	const menuButtonRef = useRef();
+	const menuRef = useRef();
+
+	useEffect( () => {
+		global.addEventListener( 'mouseup', handleMenuClose );
+		global.addEventListener( 'keyup', handleMenuClose );
+		global.addEventListener( 'keyup', handleDialogClose );
+
+		return () => {
+			global.removeEventListener( 'mouseup', handleMenuClose );
+			global.removeEventListener( 'keyup', handleMenuClose );
+			global.removeEventListener( 'keyup', handleDialogClose );
+		};
+	}, [] );
+
+	const handleMenu = useCallback( () => {
+		toggleMenu( ! menuOpen );
+	}, [ menuOpen ] );
+
+	const handleMenuClose = ( e ) => {
+		if (
+			( ( 'keyup' === e.type && 27 === e.keyCode ) || 'mouseup' === e.type ) &&
+			! menuButtonRef.current.buttonRef.current.contains( e.target ) &&
+			! menuRef.current.menuRef.current.contains( e.target )
+		) {
+			toggleMenu( false );
+		}
+	};
+
+	const handleDialog = useCallback( () => {
+		toggleDialog( ! dialogActive );
+		toggleMenu( false );
+	}, [ dialogActive ] );
+
+	const handleMenuItemSelect = useCallback( ( index, e ) => {
+		if (
+			( ( 'keydown' === e.type && (
+				13 === e.keyCode || // Enter
+					32 === e.keyCode // Space
+			) ) ||
+				'click' === e.type // Mouse
+			) ) {
```

Comment:
Let's make is slightly easier to read:

```suggestion
		if (
			( 'keydown' === e.type && ( 13 === e.keyCode || 32 === e.keyCode ) ) || // Enter or Space is pressed.
			'click' === e.type // Mouse is clicked
		) {
```